pkg/alertmanager/statefulset_test.go

```diff
@@ -1081,3 +1082,88 @@ func TestExpectStatefulSetMinReadySeconds(t *testing.T) {
 	}
 }
+
+func TestPodTemplateConfig(t *testing.T) {
+	nodeSelector := map[string]string{
+		"foo": "bar",
+	}
+	affinity := v1.Affinity{
+		NodeAffinity: &v1.NodeAffinity{},
+		PodAffinity: &v1.PodAffinity{
+			PreferredDuringSchedulingIgnoredDuringExecution: []v1.WeightedPodAffinityTerm{
+				{
+					PodAffinityTerm: v1.PodAffinityTerm{
+						Namespaces: []string{"foo"},
+					},
+					Weight: 100,
+				},
+			},
+		},
+		PodAntiAffinity: &v1.PodAntiAffinity{},
+	}
+
+	tolerations := []v1.Toleration{
+		{
+			Key: "key",
+		},
+	}
+	userid := int64(1234)
+	securityContext := v1.PodSecurityContext{
+		RunAsUser: &userid,
+	}
+	priorityClassName := "foo"
+	serviceAccountName := "alertmanager-sa"
+	hostAliases := []monitoringv1.HostAlias{
+		{
+			Hostnames: []string{"foo.com"},
+			IP:        "1.1.1.1",
+		},
+	}
+	imagePullSecrets := []v1.LocalObjectReference{
+		{
+			Name: "registry-secret",
+		},
+	}
+
+	sset, err := makeStatefulSet(&monitoringv1.Alertmanager{
+		ObjectMeta: metav1.ObjectMeta{},
+		Spec: monitoringv1.AlertmanagerSpec{
+			NodeSelector:       nodeSelector,
+			Affinity:           &affinity,
+			Tolerations:        tolerations,
+			SecurityContext:    &securityContext,
+			PriorityClassName:  priorityClassName,
+			ServiceAccountName: serviceAccountName,
+			HostAliases:        hostAliases,
+			ImagePullSecrets:   imagePullSecrets,
+		},
+	}, defaultTestConfig, "", nil)
+	if err != nil {
+		t.Fatalf("Unexpected error while making StatefulSet: %v", err)
+	}
+
+	if !reflect.DeepEqual(sset.Spec.Template.Spec.NodeSelector, nodeSelector) {
+		t.Fatalf("expected node selector to match, want %v, got %v", nodeSelector, sset.Spec.Template.Spec.NodeSelector)
+	}
+	if !reflect.DeepEqual(*sset.Spec.Template.Spec.Affinity, affinity) {
+		t.Fatalf("expected affinity to match, want %v, got %v", affinity, *sset.Spec.Template.Spec.Affinity)
+	}
+	if !reflect.DeepEqual(sset.Spec.Template.Spec.Tolerations, tolerations) {
+		t.Fatalf("expected tolerations to match, want %v, got %v", tolerations, sset.Spec.Template.Spec.Tolerations)
+	}
+	if !reflect.DeepEqual(*sset.Spec.Template.Spec.SecurityContext, securityContext) {
+		t.Fatalf("expected security context  to match, want %v, got %v", securityContext, *sset.Spec.Template.Spec.SecurityContext)
+	}
+	if sset.Spec.Template.Spec.PriorityClassName != priorityClassName {
+		t.Fatalf("expected priority class name to match, want %s, got %s", priorityClassName, sset.Spec.Template.Spec.PriorityClassName)
+	}
+	if sset.Spec.Template.Spec.ServiceAccountName != serviceAccountName {
+		t.Fatalf("expected service account name to match, want %s, got %s", serviceAccountName, sset.Spec.Template.Spec.ServiceAccountName)
+	}
+	if !reflect.DeepEqual(sset.Spec.Template.Spec.HostAliases, k8sutil.HostAliasesMonitoringV1ToCoreV1(hostAliases)) {
+		t.Fatalf("expected host aliases to match, want %s, got %s", hostAliases, sset.Spec.Template.Spec.HostAliases)
+	}
```

Comment:
To avoid calling the conversion function in the test (and couple it too much with the  implementation), I'd be fine with testing that we have the number of items on both sides and add a unit test checking the output of `HostAliasesMonitoringV1ToCoreV1()`.

```suggestion
	if len(sset.Spec.Template.Spec.HostAliases) != len(hostAliases)) {
		t.Fatalf("expected host aliases to match, want %s, got %s", len(hostAliases), len(sset.Spec.Template.Spec.HostAliases)
	}
```